nevergrad/functions/images/core.py

```diff
@@ -205,3 +200,50 @@ def _with_tag(
         assert func._initialization_kwargs is not None
         func._initialization_kwargs["tags"] = tags
         return func
+
+
+class ImageFromPGAN(base.ExperimentFunction):
+    """
+    Creates face images using a GAN from pytorch GAN zoo trained on celebAHQ and optimizes the noise vector of the GAN
+
+    problem_name: the type of problem we are working on.
+    initial_noise: the initial noise of the GAN. It should be of dimension (1, 512). If None, it is defined randomly.
+    use_gpu: whether to use gpus to compute the images
+    scorer: which scorer to use for the images
+    mutable_sigma: whether the sigma should be mutable
+    n_mutations: number of mutations
+    """
+
+    def __init__(self, problem_name: str = "GAN_optimization", initial_noise: np.ndarray = None, use_gpu: bool = True, scorer=Koncept512Loss, mutable_sigma=True, n_mutations=35) -> None:
+        if torch.cuda.is_available():
+            use_gpu = False
+
+        # Storing high level information..
+        self.pgan_model = torch.hub.load('facebookresearch/pytorch_GAN_zoo:hub',
+                               'PGAN', model_name='celebAHQ-512',
+                               pretrained=True, useGPU=use_gpu)
+
+        self.problem_name = problem_name
+        self.noise_shape = (1, 512)
+        if initial_noise is None:
+            initial_noise = np.random.normal(size=self.noise_shape)
+        assert initial_noise.shape == self.noise_shape, f'The shape of the initial noise vector was {initial_noise.shape}, it should be {self.noise_shape}'
+
+        array = ng.p.Array(init=initial_noise, mutable_sigma=mutable_sigma)
+        # parametrization
+        array.set_mutation(sigma=n_mutations)
+        array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1))).set_name("")
+
+        super().__init__(self._loss, array)
+        self.register_initialization(problem_name=problem_name)
```

Comment:
this function is used to create a copy of the instance by reinitializing a new instance with the parameters you provide here. Since the useful parameters are all but the problem name, the new copy may be completely different from the initial version, so you'll need to update this

Reply:
actually, if we wait for #914 then this won't be needed anymore :D

Reply:
should I use a name that contains the initial noise ? For now I used the name using the other parameters

Reply:
I don't understand what you mean. In any case for now I guess you can just remove it

Reply:
You can just remove register_initialization, given jeremy's PR (provided that you git pull origin master).